src/modal/modal-stack.ts

```diff
@@ -47,10 +54,12 @@ export class NgbModalStack {
 
     this._registerModalRef(ngbModalRef);
     ngbModalRef.result.then(revertPaddingForScrollBar, revertPaddingForScrollBar);
+    ngbModalRef.result.then(removeBodyClass, removeBodyClass);
     activeModal.close = (result: any) => { ngbModalRef.close(result); };
     activeModal.dismiss = (reason: any) => { ngbModalRef.dismiss(reason); };
 
     this._applyWindowOptions(windowCmptRef.instance, options);
+    renderer.addClass(this._document.body, 'modal-open');
```

Comment:
Could we check if there are no open modals open already and skip DOM manipulation if there is already an open modal?

Reply:
@pkozlowski-opensource why do you think this is necessary? How is it possible to open a modal without using NgbModalStack.open method?

Reply:
Done. `addClass` will be called only when `_modalRefs` has 1 modal.